src/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/common/ResultCode.cs

```diff
@@ -57,61 +55,59 @@ public enum ResultCode
 
     internal class OperationErrorMappings
     {
-        private static Hashtable s_resultCodeHash = null;
+        private static Dictionary<ResultCode, string> s_resultCodeHash;
 
         static OperationErrorMappings()
```

Comment:
Doesn't seem like this needs to be an explicit static cctor.  You could make s_resultCodeHash readonly and just initialize it inline.  (You could also add a capacity to the Dictionary's ctor that matches the number if items you'll be adding to it.)